worlds/alttp/Shops.py

```diff
@@ -207,10 +207,10 @@ def ShopSlotFill(world):
             shops_per_sphere.append(current_shops_slots)
             candidates_per_sphere.append(current_candidates)
             for location in sphere:
-                if location.shop_slot is not None:
+                if isinstance(location, ALttPLocation) and location.shop_slot is not None:
                     if not location.shop_slot_disabled:
                         current_shops_slots.append(location)
```

Comment:
Since shop_slot_disabled = False is in the class body, how about
```suggestion
                if isinstance(location, ALttPLocation) and if not location.shop_slot_disabled:
                    current_shops_slots.append(location)
```

Reply:
While i do not understand the if-tree (especially the elseif part), the suggested change changes behaviour.

Currently if it's a shop_slot and it's disabled, the location is skipped. After the change it would still be used if the location is not locked and not excluded by name.